### ISSUE TITLE
🐛 [firebase_in_app_messaging] Failing pod install

### ISSUE THREAD

**Issue by subair-zufi** [firebase/flutterfire#11924]
**Failing pod install after adding firebase_in_app_messaging to project**
```
firebase_messaging: ^14.7.5
firebase_core: ^2.23.0
firebase_in_app_messaging: ^0.7.4+5
firebase_database: ^10.3.5
firebase_core_platform_interface: ^5.0.0
```

***Result***
```
Analyzing dependencies
firebase_core: Using Firebase SDK version '10.18.0' defined in 'firebase_core'
firebase_database: Using Firebase SDK version '10.18.0' defined in 'firebase_core'
firebase_in_app_messaging: Using Firebase SDK version '10.18.0' defined in 'firebase_core'
firebase_messaging: Using Firebase SDK version '10.18.0' defined in 'firebase_core'
[!] CocoaPods could not find compatible versions for pod "Firebase/Database":
  In snapshot (Podfile.lock):
    Firebase/Database (= 10.2.0)

  In Podfile:
    firebase_database (from `.symlinks/plugins/firebase_database/ios`) was resolved to 10.3.5, which depends on
      Firebase/Database (= 10.18.0)


You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * changed the constraints of dependency `Firebase/Database` inside your development pod `firebase_database`.
   You should run `pod update Firebase/Database` to apply changes you've made.

```

***flutter doctor***
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.16.0, on macOS 13.2 22D49 darwin-arm64, locale en-IN)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✗] Chrome - develop for the web (Cannot find Chrome executable at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.
[✓] Android Studio (version 2021.3)
[✓] IntelliJ IDEA Community Edition (version 2022.3.2)
[✓] VS Code (version 1.84.2)
[✓] Connected device (2 available)
[✓] Network resources

! Doctor found issues in 1 category.

```

**Comment by danagbemava-nc** [comment]
Hi @subair-zufi, can you enter your iOS directory and run `pod install --repo-update` to see if that fixes the issue?

**Comment by subair-zufi** [comment]
It fails @danagbemava-nc,
Logs for `pod install --repo-update`:

```
subairk@Subairs-Air ios % pod install --repo-update
Updating local specs repositories
  $ /usr/bin/git -C /Users/subairk/.cocoapods/repos/cloudsmith-kzflb1ryyncxnqr9-credolab-proxyen-sdk-cocoapods-index fetch origin --progress
  $ /usr/bin/git -C /Users/subairk/.cocoapods/repos/cloudsmith-kzflb1ryyncxnqr9-credolab-proxyen-sdk-cocoapods-index rev-parse --abbrev-ref HEAD
  master
  $ /usr/bin/git -C /Users/subairk/.cocoapods/repos/cloudsmith-kzflb1ryyncxnqr9-credolab-proxyen-sdk-cocoapods-index reset --hard origin/master
  HEAD is now at 9f17ca9 Added CredoAppCalendarEvents 3.1.0 to index
Analyzing dependencies
firebase_core: Using Firebase SDK version '10.18.0' defined in 'firebase_core'
firebase_database: Using Firebase SDK version '10.18.0' defined in 'firebase_core'
firebase_in_app_messaging: Using Firebase SDK version '10.18.0' defined in 'firebase_core'
firebase_messaging: Using Firebase SDK version '10.18.0' defined in 'firebase_core'
[!] CocoaPods could not find compatible versions for pod "Firebase/Database":
  In snapshot (Podfile.lock):
    Firebase/Database (= 10.2.0)

  In Podfile:
    firebase_database (from `.symlinks/plugins/firebase_database/ios`) was resolved to 10.3.5, which depends on
      Firebase/Database (= 10.18.0)


You have either:
 * changed the constraints of dependency `Firebase/Database` inside your development pod `firebase_database`.
   You should run `pod update Firebase/Database` to apply changes you've made.

```

**Comment by danagbemava-nc** [comment]
Hi @subair-zufi, sorry for the late reply. 

Have you tried running `pod update Firebase/Database` to see if it helps? 

Can you also try deleting your `Podfile.lock` and re-running `pod install` to see if you still get the issue?
